api/src/main/java/ca/bc/gov/educ/api/batchgraduation/reader/RegenerateCertificatePartitioner.java

```diff
@@ -43,37 +42,39 @@ public Map<String, ExecutionContext> partition(int gridSize) {
             if(parallelDTO != null) {
                 credentialList.addAll(parallelDTO.certificateList());
             }
+        } else if (certificateRegenerationRequest.getPens() != null && !certificateRegenerationRequest.getPens().isEmpty()) {
+            certificateRegenerationRequest.getPens().forEach(p -> {
+                StudentCredentialDistribution scd = new StudentCredentialDistribution();
+                scd.setPen(p);
+                scd.setStudentID(getStudentIDByPen(p));
```

Comment:
getStudentIDByPen() might return null here. Does design allow for nulls? Would this cause an error downstream?

Reply:
No, null studentID will be skipped in batch processor.